### PR TITLE
Refactor requests in database to support timestamps

### PR DESCRIPTION
**Pull request by akskokki**
Requests now have their own table, and all relevant parts of the program have been refactored to support this change.

Each Request connects one User to one Course, and stores a timestamp of the moment the request was made.

### CODE REVIEW

File: src/lib/prisma/users.ts [hunk 1]
```diff
@@ -134,16 +116,20 @@ export async function getUsersEnrollsAndRequests(userId: string) {
...
 
+export async function getUsersRequests(userId: string) {
+  const requests = await prisma.request.findMany({
+    where: {
+      userId: userId,
+    },
+  });
+  return requests;
+}
+
```
Comment (erjavaskivuori) [inline src/lib/prisma/users.ts:132]: Not a big thing but I think this function should be in the requests.ts file as the fetch is done from the request table

**Review by erjavaskivuori — CHANGES_REQUESTED** [review]
Commented about location of one function but otherwise looking good to me!

File: src/lib/prisma/users.ts (hunk 1 above)
Comment (akskokki) [reply src/lib/prisma/users.ts:132]: The reason the function is in this file is that it only returns data from the users table, we talked about it and agreed that it makes sense to keep it here

**Review by erjavaskivuori — APPROVED** [review]
:shipit:

**Pull request merged**